http.go

```diff
@@ -40,9 +41,17 @@ type ServerHandler struct {
 	modifiedWRPCount         metrics.Counter
 	incomingQueueDepth       int64
 	maxOutstanding           int64
+	incomingQueueLatency     metrics.Histogram
+	now                      func() time.Time
 }
 
 func (sh *ServerHandler) ServeHTTP(response http.ResponseWriter, request *http.Request) {
+	eventType := unknownEventType
+	// find time difference, add to metric after function finishes
+	defer func() {
+		sh.recordQueueLatencyToHistogram(sh.now(), eventType)
+	}()
```

Comment:
don't we want to execute `sh.now()` outside of the defer, otherwise it'll be execute at the end of `ServeHTTP` and not at the beginning?
like so
```go
defer func(s) {
	sh.recordQueueLatencyToHistogram(s, eventType)
}(sh.now())
```